kms/src/main/java/com/example/Snippets.java

```diff
@@ -0,0 +1,448 @@
+        .execute();
+
+    System.out.println(newIamPolicy);
+    return newIamPolicy;
+  }
+
+  /**
+   * Adds the given member to the given keyring, with the given role.
+   *
+   * @param ringId The id of the keyring.
+   * @param member The member to add. Must be in the proper format, eg:
+   *
+   *        allUsers
+   *        user:$userEmail
+   *        serviceAccount:$serviceAccountEmail
+   *
+   *        See https://g.co/cloud/kms/docs/reference/rest/v1beta1/Policy#binding
+   *        for more details.
+   *
+   * @param role Must be in one of the following formats:
+   *        roles/[role]
+   *        organizations/[organizationId]/roles/[role]
+   *        projects/[projectId]/roles/[role]
+   *
+   *        See https://g.co/cloud/iam/docs/understanding-roles
+   *        for available values for [role].
+   */
+  public static Policy addToKeyRingPolicy(
+      String projectId, String ringId, String member, String role)
+      throws IOException {
+    String location = "global";
+    // Create the Cloud KMS client.
+    CloudKMS kms = createAuthorizedClient();
+
+    // The resource name of the keyring version
+    String keyring = String.format(
+        "projects/%s/locations/%s/keyRings/%s",
+        projectId, location, ringId);
+
+    // Get the current IAM policy
+    Policy iamPolicy = getKeyRingPolicy(projectId, ringId);
+
+    // Add the new account to it.
+    Binding newBinding = new Binding()
+        .setRole(role)
+        .setMembers(Collections.singletonList(member));
+    List<Binding> bindings = iamPolicy.getBindings();
+    if (null == bindings) {
+      bindings = Collections.singletonList(newBinding);
+    } else {
+      bindings.add(newBinding);
+    }
+    iamPolicy.setBindings(bindings);
+
+    // Set the new IAM Policy.
+    Policy newIamPolicy = kms.projects().locations().keyRings()
+        .setIamPolicy(keyring, new SetIamPolicyRequest().setPolicy(iamPolicy))
+        .execute();
+
+    System.out.println(newIamPolicy);
+    return newIamPolicy;
+  }
+
+  /**
+   * Removes the given member from the given policy.
+   */
+  public static Policy removeFromCryptokeyPolicy(
+      String projectId, String ringId, String keyId, String member, String role)
+      throws IOException {
+    String location = "global";
+    // Create the Cloud KMS client.
+    CloudKMS kms = createAuthorizedClient();
+
+    // The resource name of the cryptoKey
+    String cryptoKey = String.format(
+        "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
+        projectId, location, ringId, keyId);
+
+    // Get the current IAM policy and add the new account to it.
+    Policy iamPolicy = getCryptoKeyPolicy(projectId, ringId, keyId);
+
+    List<Binding> bindings = iamPolicy.getBindings();
+    // Filter out the given member
+    for (Binding b : bindings) {
+      if (role.equals(b.getRole()) && b.getMembers().contains(member)) {
+        b.getMembers().remove(member);
+        /*
```

Comment:
Ah. Right.
Since `.remove` can be optionally implemented, I wasn't sure if the `List` returned by `getMembers()` had it, so this was the contingency plan. Turns out it does, so this works, but I forgot I had the contingency here :-) Removed.